k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/selectedstate/command/ImapCommandSplitter.java

```diff
@@ -0,0 +1,106 @@
+package com.fsck.k9.mail.store.imap.selectedstate.command;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+
+import com.fsck.k9.mail.store.imap.selectedstate.command.SelectedStateCommand.Builder;
+import com.fsck.k9.mail.store.imap.selectedstate.command.SelectedStateCommand.ContiguousIdGroup;
+
+
+class ImapCommandSplitter {
+
+    static List<SelectedStateCommand> splitCommand(SelectedStateCommand command, int lengthLimit) {
```

Comment:
Yes. Nicely encapsulated functionality like this should definitely be tested individually. It's also nice being able to jump to the tests for a class with `Ctrl+Shift+T` if the name matches